### PR TITLE
Dashboard using optimizing visualizer.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

**Comment by dkorolev** [comment]
@mzhurovich : And here's the real-life optimizer code!

@deathbaba @sompylasar : FYI.

File: demo.cc [hunk 1]
```diff
@@ -202,8 +217,13 @@ class Cruncher final {
   struct Consumer {
     const std::string& demo_id_;
     Box box_;
+
+    std::string current_image_ = "NO DATA YET\n";
```
Comment (sompylasar) [inline demo.cc]: WTF? Maybe an empty 1px gif in a data-URI?

Comment (sompylasar) [inline demo.cc]: Or an empty string?

**Comment by sompylasar** [comment]
Cool! I'll try to get this running.

File: demo.cc (hunk 1 above)
Comment (dkorolev) [inline demo.cc]: Hey, it's a first prototype.

By the time we launch a hosted demo (a couple days?), the could will ensure the first "real" image is generated before the user can request it.

Comment (sompylasar) [inline demo.cc]: I mean, this had better be of the same type with the data (e.g. a URL).

**Comment by dkorolev** [comment]
Guys, please take another look.

@sompylasar : Fixed the default image.

**Comment by sompylasar** [comment]
@dkorolev Thanks, looks better.

**Comment by dkorolev** [comment]
Okay, I'm merging this one now, since we have three pending in total ...

**Pull request merged**